include/irr/builtin/glsl/bxdf/brdf/cos_weighted_sample.glsl

```diff
@@ -13,14 +13,27 @@ irr_glsl_BSDFSample irr_glsl_cos_weighted_cos_gen_sample(in irr_glsl_Anisotropic
 
     irr_glsl_BSDFSample smpl;
     smpl.L = L;
-    smpl.probability = dot(interaction.N,smpl.L)*irr_glsl_RECIPROCAL_PI;
+	smpl.LdotN = dot(interaction.N,smpl.L);
+	//TODO fill other smpl.xxxxx
 
     return smpl;
 }
-irr_glsl_BSDFSample irr_glsl_cos_weighted_cos_gen_sample(in irr_glsl_AnisotropicViewSurfaceInteraction interaction, in uvec2 _sample)
+irr_glsl_BSDFSample irr_glsl_cos_weighted_cos_generate(in irr_glsl_AnisotropicViewSurfaceInteraction interaction, in uvec2 _sample)
 {
     vec2 u = vec2(_sample)/float(UINT_MAX);
-    return irr_glsl_cos_weighted_cos_gen_sample(interaction, u);
+    return irr_glsl_cos_weighted_cos_generate(interaction, u);
+}
+
+//returning just 1.0 since we dont know which brdf is being sampled, maybe return irr_glsl_PI instead
+//TODO oren-nayar cos_eval with rec_pi factored out
```

Comment:
sampling functions for BRDFs are only possible for physically based distributions (ones that are normalized etc) so only provide for the PBR distribution

Reply:
hm? not sure what is this comment about

Reply:
basically oren-nayar with the PI factored out is not a PBR distribution, not using a PBR distribution for sample generation (which you only ever use for IBL or Path Tracing) doesn't make sense as per-our MineGameDev presentation.

Reply:
are we clear?

Reply:
i dont remember why i wanted to factor PI out, but i dont think i wanted to sample oren-nayar without pi

Reply:
lets just not do it ;)